CMakeLists.txt

```diff
@@ -154,13 +154,48 @@ else()
     endif()
   endif()
 
-  install(TARGETS nanoarrow DESTINATION lib)
+  install(TARGETS nanoarrow
+          DESTINATION lib
+          EXPORT nanoarrow-exports)
+  install(TARGETS coverage_config
+          DESTINATION lib
+          EXPORT nanoarrow-exports)
```

Comment:
That's what the [NAMESPACE](https://cmake.org/cmake/help/latest/command/install.html#export) is for, it is considered good practice to use namespaced targets like `nanoarrow::coverage_config` even within projects (which requires creating an ALIAS) but def for exported targets. We should use that here as well.

Reply:
I agree, this would be the right solution. I didn't do it initially because 1) I was worried about breaking existing users and 2) I didn't know how far the scope of this PR should go. In hindsight, point 1 is moot because this package wasn't previously importable anyway, so even with adding the namespace existing consumers would still see the un-namespaced targets. For point 2, IMO best practice would be to update all the examples/code within the repo to use the namespaced target consistently whether they're getting nanoarrow via FetchContent or `find_package`. However, I think that's probably out of scope for this PR and can be done in some future work. Would you agree?

Reply:
Also for context @paleolimbot this is how I would anticipate consuming the components that I suggested in #407. You'd produce an importable nanoarrow that has targets like `nanoarrow::nanoarrow`, `nanoarrow::coverage_config`, `nanoarrow::ipc`, `nanoarrow::device`, and then the user could do something like `find_package(nanoarrow COMPONENTS device)` and then `target_link_libraries(my_pkg PRIVATE nanoarrow_device)`.

Reply:
I removed the target and just set the options as public on the nanoarrow target so that they'll propagate to everything linking to it. That should be sufficient and avoids the weirdness of this target.